### PR TITLE
feature(cluster_docker): support using specific docker network

### PR DESCRIPTION
**Pull request by fruch**
now all nodes are created on the default bridge network
but for some cases we have components using docker-compose
which by default creates it's own network, we can communicate
to our nodes from those docker-compose based setups.

so this new configuration option, would enable to tell upfront
that we need to create the nodes in a specific network.

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: sdcm/cluster.py [hunk 1]
```diff
@@ -5124,12 +5124,13 @@ def node_setup(self, node, **kwargs):  # pylint: disable=unused-argument
...
 
         if self.test_config.REUSE_CLUSTER:
-            node.disable_daily_triggered_services()
+
             self.configure_scylla_monitoring(node)
             self.restart_scylla_monitoring(sct_metrics=True)
             set_grafana_url(f"http://{normalize_ipv6_url(node.external_address)}:{self.grafana_port}")
             return
 
+        node.disable_daily_triggered_services()
```
Comment (vponomaryov) [inline sdcm/cluster.py:5133]: This change doesn't fit the `mocking` part. It is change of logic for other backends.
If it is intentional then we have bug now? Or why is it changed?

**Comment by soyacz** [comment]
doesn't it break monitoring?

File: sdcm/utils/docker_utils.py [hunk 2]
```diff
@@ -309,9 +309,11 @@ def wait_for_status(cls, instance: INodeWithContainerManager, name: str, status:
     @retrying(n=10, sleep_time=1, allowed_exceptions=(Retry, ))
     def get_ip_address(cls, instance: INodeWithContainerManager, name: str) -> str:
         ip_address = cls.get_container(instance, name).attrs["NetworkSettings"]["IPAddress"]
-        if not ip_address:
+        network_ip_address = cls.get_container(
+            instance, name).attrs["NetworkSettings"]["Networks"][instance.parent_cluster.params.get('docker_network')]["IPAddress"]
```
[the remark this answers is not in the document]
Comment (fruch) [reply sdcm/utils/docker_utils.py:313]: Yep in the following commit doing the same for integration tests, I did exactly that

**Comment by fruch** [comment]
> doesn't it break monitoring?

You mean it might break monitoring.
Cause of the call that was moved ?

Or the docker networking setup?

File: sdcm/cluster.py (hunk 1 above)
Comment (fruch) [reply sdcm/cluster.py:5132]: now on it's own commit

**Comment by fruch** [comment]
> > > doesn't it break monitoring?
> > 
> > 
> > You mean it might break monitoring. Cause of the call that was moved ?
> > Or the docker networking setup?
> 
> docker networking setup - prometheus will try to get to node exporter/scylla metrics and I think we don't forward these ports

it might, if it doesn't setup with host networking

**Review by vponomaryov — APPROVED** [review]
LGTM

**Pull request merged**